client/src/main/scala/org/http4s/client/websocket/WSClient.scala

```diff
@@ -133,42 +139,67 @@ private[http4s] trait WSConnectionHighLevel[F[_]] {
   def sendMany[G[_]: Foldable, A <: WSDataFrame](wsfs: G[A]): F[Unit]
 
   /** A `Pipe` which sends websocket frames and emits a `()` for each chunk sent. */
-  final def sendPipe: Pipe[F, WSDataFrame, Unit] = _.chunks.evalMap(sendMany(_))
-
-  /** Send a Close frame. The sending side of this connection will be closed. */
-  def sendClose(reason: String = ""): F[Unit]
+  def sendPipe: Pipe[F, WSDataFrame, Unit] = _.chunks.evalMap(sendMany(_))
 
   /** Wait for a websocket frame to be received. Returns `None` if the receiving side is closed.
     * Fragmentation is handled automatically, the `last` attribute can be ignored.
     */
   def receive: F[Option[WSDataFrame]]
 
   /** A stream of the incoming websocket frames. */
-  final def receiveStream: Stream[F, WSDataFrame] = Stream.repeatEval(receive).unNoneTerminate
+  def receiveStream: Stream[F, WSDataFrame] = Stream.repeatEval(receive).unNoneTerminate
 
   /** The negotiated subprotocol, if any. */
-  def subprocotol: Option[String]
+  def subprotocol: Option[String]
 
   /** The close frame, if available. */
   def closeFrame: TryableDeferred[F, WSFrame.Close]
+
+  def mapK[G[_]](fk: F ~> G): WSConnectionHighLevel[G] =
+    new WSConnectionHighLevel[G] {
+      def send(wsf: WSDataFrame): G[Unit] = fk(outer.send(wsf))
+      def sendMany[H[_]: Foldable, A <: WSDataFrame](wsfs: H[A]): G[Unit] = fk(outer.sendMany(wsfs))
+      def receive: G[Option[WSDataFrame]] = fk(outer.receive)
+      def subprotocol: Option[String] = outer.subprotocol
+      def closeFrame: TryableDeferred[G, WSFrame.Close] = outer.closeFrame.mapK(fk)
```

Comment:
This is where we get into trouble. Unfortunately `TryableDeferred#mapK` returns a `Deferred`, not a `TryableDeferred`. In principle we could upstream a fix to CE2 😩